app/src/org/commcare/models/database/user/models/AndroidCaseIndexTable.java

```diff
@@ -279,4 +291,17 @@ public static String getArgumentBasedVariableSet(int number) {
         return sb.toString();
     }
 
+    public void reIndexAllCases(SqlStorage<ACase> caseStorage) {
+        db.beginTransaction();
+        try {
+            wipeTable();
+            for(ACase c : caseStorage) {
+                indexCase(c);
```

Comment:
Oh, sorry,t o be clear: wipeTable() does delete rows, not the full table. db.delete(TABLE_NAME, null, null); is "delete all rows", not "Drop table"